### DISCUSSION TITLE
C# markup: how far could we take it?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#14383] [Q&A] [answered]
Ever since I started my UNO project many years back, I avoided XAML but have been using my custom C# markup instead, which you could consider as a very simplified version of what UNO 5 now supports officially or what Vincent Hoogendorn came up with several years back.

Right from the start I felt there still is one area where I needed XAML: this is the templating - in my case a custom version of the UNO Material templating.

What is UNO's official stance on the styling/templating (Material, Fluent, etc.)? Does UNO consider this as being 'replaceble' by C# (markup) as well or do we still have to go with XAML for the time being?

Note: I have a few DataTemplates which I needed to instantiate by XAML + XamlReader.Load ... but that's a different story ...

**Comment by MartinZikmund** [comment, marked as answer]
Both styles and templates are supported by C# markup, see the following docs:

- [Styles](https://platform.uno/docs/articles/external/uno.extensions/doc/Reference/Markup/Styles.html)
- [Templates](https://platform.uno/docs/articles/external/uno.extensions/doc/Reference/Markup/Templates.html)

**Reply by DierkDroth** [reply]
Thanks for clarification @MartinZikmund
